### PR TITLE
Fix static file caching issue

### LINKED ISSUE #136: Cache issue for deployed SPA updates
### Expected Behaviour

An end-user should always access the latest deployment of a Project Firefly SPA from the browser. 
This means that the browser should load the latest version of the static files from the cache. 

The TTLs are defined as follow:

- 1 min for HTML
- 7 days for every other mime type

The Parcel bundler generates content hashes for the static files, which are used as selectors to ensure the expected behavior.

### Actual Behaviour

The Parcel bundler doesn't generate new hashes upon static files updates, which forces the end-user to manually clean the browser cache in order to use the latest SPA deployment.

### Reproduce Scenario (including but not limited to)

- Create a new SPA using the `aio CLI`
- Deploy the SPA using `aio app deploy`
- Access the deployed SPA from your browser
- Update the SPA front-end code (e.g. update some JS or CSS file)
- Redeploy the SPA using `aio app deploy`
- Even after one minute delay and page refresh, the deployed changes are not loaded in the browser
- Manually clean the cache of your browser or access the deployed SPA from another browser
- The deployed changes are loaded in the browser

### PR DESCRIPTION
**Pull request by sandeep-paliwal**
Content hash was not included in file names when parcel bundle was generated. This created same file name for static content which was referenced in HTML file. With same file name cached version of file was used and latest changes were not reflected to user.

## Description
Fix is to use the content hash option provided by parcel bundler.

## How Has This Been Tested?
manual app deploy
run unit tests

## Screenshots (if appropriate):

Fixes #136

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by meryllblanchet** [comment]
Fixes #136

**Comment by shazron** [comment]
A few issues:
1. Please update the CI to use GIthub Actions. see aio-cli for the [file](https://github.com/adobe/aio-cli/blob/master/.github/workflows/node.js.yml). Make sure in the repo settings Github Actions is enabled as well.
2. Remove the .travis.yml file
3. Remove the Travis CI build badge in the README.md
4. Add the Github Actions build badge in the README.md (see aio-cli [README](https://raw.githubusercontent.com/adobe/aio-cli/master/README.md))
5. Add a test to see if the bundler options in the build-web call will always be called with `contentHash: true`
6. Branch protection for master branch in repo settings -> check the appropriate checkboxes for the new CI tests. Note: they will only show up after the first build is run

**Comment by moritzraho** [comment]
LGTM,
Checks done:
- chrome: hitting the refresh button after a new deployment now refreshes content while it wasn't before 👌
- firefox: no noticeable changes (seems my FF is not caching js and css files)
- `aio app run`: checked that content hash is not applied here as it would break watch 👌 
- `aio app deploy`: made sure that previous files are deleted, as otherwise old static files would pile up

**Comment by sandeep-paliwal** [comment]
@shazron I will create a separate PR for the CI changes you mentioned.

**Comment by shazron** [comment]
> @shazron I will create a separate PR for the CI changes you mentioned.

thanks. That PR has to be merged in and those changes merged into this PR for this PR to pass CI.

File: test/src/build-web.test.js [hunk 1]
```diff
@@ -48,4 +48,20 @@ describe('build-web', () => {
...
+      },
+      web: {
+        distProd: 'dist',
+        src: 'fakeDir'
+      }
+    }
+    global.addFakeFiles(vol, 'fakeDir', { 'index.html': '' })
+    fs.readdir.mockReturnValue(['output.html'])
+    await expect(buildWeb(config)).resolves.toEqual(['output.html'])
+    expect(global._bundler__arguments).toEqual(['fakeDir/index.html', { cache: false, contentHash: true, logLevel: 0, outDir: 'dist', publicUrl: './', watch: false }])
```
Comment (shazron) [inline test/src/build-web.test.js:65]: forward slash causing error in Windows tests. I'll fix

**Review by shazron — CHANGES_REQUESTED** [review]

**Review by shazron — APPROVED** [review]

**Comment by alexkli** [comment]
FWIW, I can confirm that this is fixed in `@adobe/aio-cli 5.0.1`.

**Pull request merged**